regparse.c

```diff
@@ -6086,10 +6089,55 @@ node_extended_grapheme_cluster(Node** np, ScanEnv* env)
           R_ERR(create_node_from_array(LIST, core_alts+4, XP_list));
         }
 
+        /* conjunctCluster := \p{InCB=Consonant} ([\p{InCB=Extend} \p{InCB=Linker}]* \p{InCB=Linker} [\p{InCB=Extend} \p{InCB=Linker}]* \p{InCB=Consonant})+ */
+        {
+          Node **CC_list = core_alts + 6; /* size: 2 */
+          R_ERR(create_property_node(CC_list+0, env, "InCB_Consonant"));
+
+          {
+            Node **CC_list1 = CC_list + 1; /* size: 4 */
+            // [\p{InCB=Extend} \p{InCB=Linker}]*
+            {
+              {
+                Node **CC_alt1 = CC_list1 + 1;
+                R_ERR(create_property_node(CC_alt1+1, env, "InCB_Extend"));
+                R_ERR(create_property_node(CC_alt1+2, env, "InCB_Linker"));
+
+                R_ERR(create_node_from_array(ALT, CC_alt1, CC_alt1+1));
+              }
+
+              R_ERR(quantify_node(CC_list1+1, 0, REPEAT_INFINITE));
+            }
+
+            // \p{InCB=Linker}
+            R_ERR(create_property_node(CC_list1+2, env, "InCB_Linker"));
+
+            // [\p{InCB=Extend} \p{InCB=Linker}]*
+            {
+              {
+                Node **CC_alt2 = CC_list1 + 3;
+                R_ERR(create_property_node(CC_alt2+1, env, "InCB_Extend"));
+                R_ERR(create_property_node(CC_alt2+2, env, "InCB_Linker"));
+
+                R_ERR(create_node_from_array(ALT, CC_alt2, CC_alt2+1));
+              }
+
+              R_ERR(quantify_node(CC_list1+3, 0, REPEAT_INFINITE));
+            }
+
+            // \p{InCB=Consonant}
+            R_ERR(create_property_node(CC_list1+4, env, "InCB_Consonant"));
+
+            R_ERR(create_node_from_array(LIST, CC_list1, CC_list1+1));
+            R_ERR(quantify_node(CC_list1, 1, REPEAT_INFINITE));
+          }
+          R_ERR(create_node_from_array(LIST, core_alts+5, CC_list));
+        }
+
         /* [^Control CR LF] */
-        core_alts[5] = node_new_cclass();
-        if (IS_NULL(core_alts[5])) goto err;
-        cc = NCCLASS(core_alts[5]);
+        core_alts[6] = node_new_cclass();
+        if (IS_NULL(core_alts[6])) goto err;
+        cc = NCCLASS(core_alts[6]);
```

Comment:
This part is covered by an automatically generated test in this file based on GraphemeBreakTest.txt.

https://github.com/ruby/ruby/blob/dfc25204235079e23eadf9e0ba860c1ebcb14325/test/ruby/enc/test_grapheme_breaks.rb
https://www.unicode.org/Public/15.1.0/ucd/auxiliary/GraphemeBreakTest.txt